packages/prettier-plugin-liquid/src/printer/print/liquid.ts

```diff
@@ -165,6 +165,12 @@ function printNamedLiquidBlockStart(
       ]);
     }
 
+    case NamedTags.content_for: {
+      const markup = node.markup;
+      const trailingWhitespace = markup.args.length > 0 ? line : ' ';
+      return tag(trailingWhitespace);
+    }
```

Comment:
The `tag(trailingWhitespace)` helper is here to handle the logic about what "whitespace-logic-char" we want to use at the end of the tag.

```
// when it doesn't break
tag('') => {% content_for "blocks"%}
tag(' ') => {% content_for "blocks" %}
tag(line) => {% content_for "blocks" %}

// when it breaks
tag('') => {% content_for "blocks"%}
tag(' ') => {% content_for "blocks" %}
tag(line) => {% content_for "blocks"
%}
```